### PR TITLE
Refactored social_graphql according to phpstan feedback and updated baseline for PR #2770

### PR DESCRIPTION
**Pull request by navneet0693**
## Problem
PHPStan fixes for https://github.com/goalgorilla/open_social/pull/2770

## Solution
1. Comment out phpstan-baseline in phpstan.neon file.
2. Analyse using PHPStan
3. Refactor the code according to feedback.

## Issue tracker
N.A

## How to test
- [ ] All checks should pass.

## Screenshots
N.A

## Release notes
N.A

## Change Record
N.A

## Translations
N.A

### CODE REVIEW

**Comment by open-social-tugboat** [comment]
[Tugboat](https://tugboat.qa) has finished building the preview for this pull request!

Link:
- https://pr2788-v3kymdwchfdleqbhyormgx6bwgvwe1yb.tugboat.qa


Dashboard:
- https://dashboard.tugboat.qa/6217b9abdf1c1f9d5eb29061

File: modules/custom/social_graphql/src/GraphQL/EntityConnection.php [hunk 1]
```diff
@@ -182,6 +182,9 @@ public function nodes() : SyncPromise {
    * returning edges or nodes, this needs to be compensated in the same way.
    * This function removes the overfetching and ensures the results are in the
    * requested order.
+   *
+   * @return \GraphQL\Executor\Promise\Adapter\SyncPromise
+   *   The edges.
```
Comment (Kingdutch) [inline modules/custom/social_graphql/src/GraphQL/EntityConnection.php:187]:
```suggestion
   *   The result in the order as requested by the client which may differ from how we get it from the database in `getResult`.
```

Linebreaks may need to be added for coding standards.

File: modules/custom/social_graphql/src/GraphQL/EntityConnection.php [hunk 2]
```diff
@@ -234,10 +237,10 @@ protected function hasResult() : bool {
    *   The result for this connection's query.
    */
   protected function getResult() : SyncPromise {
-    if (!$this->hasResult()) {
-      $this->result = $this->execute();
+    if ($this->result instanceof SyncPromise) {
+      return $this->result;
     }
-    return $this->result;
+    return $this->execute();
```
Comment (Kingdutch) [inline modules/custom/social_graphql/src/GraphQL/EntityConnection.php:243]: 🚨  This change breaks the code.

I'm also unclear on why the change is needed.

File: modules/custom/social_graphql/src/GraphQL/EntityConnection.php [hunk 3]
```diff
@@ -247,6 +250,7 @@ protected function getResult() : SyncPromise {
    *   A promise that resolves to the edges of this connection.
    */
   protected function execute() : SyncPromise {
+    /** @var \Drupal\Core\Entity\Query\QueryBase $query */
     $query = $this->queryHelper->getQuery();
```
Comment (Kingdutch) [inline modules/custom/social_graphql/src/GraphQL/EntityConnection.php:254]: [Prefer assertions](https://phpstan.org/writing-php-code/phpdocs-basics#inline-%40var) that break our tests when things are broken, rather than annotations which don't. Assertions are treated like comments in productions.

```suggestion
    $query = $this->queryHelper->getQuery();
    assert($query instanceof QueryBase, "The query helper getQuery method didn't return an entity query. This is an implementation error");
```

File: modules/custom/social_graphql/src/GraphQL/StandardisedMutationSchemaTrait.php [hunk 4]
```diff
@@ -37,20 +37,23 @@ trait StandardisedMutationSchemaTrait {
...
-    );
+      $registry->addFieldResolver('Mutation', $field_name,
+        $builder->compose(
+          $builder->produce($data_producer . "_input")
+            ->map('input', $builder->fromArgument('input')),
+          $builder->produce($data_producer)
+            ->map('input', $builder->fromParent())
+        )
+      );
+    }
```
Comment (Kingdutch) [inline modules/custom/social_graphql/src/GraphQL/StandardisedMutationSchemaTrait.php:56]: This doesn't handle the case where `$result` is not an error. However, that case no longer registers a field resolver so I'd consider it a fatal error. 

How can we improve this code so that either:

A) The code never fails
B) Developers are notified of the error?

To help us with the above: when is `$result` not an array?

File: modules/custom/social_graphql/src/Plugin/GraphQL/DataProducer/Payload/PayloadViolations.php [hunk 5]
```diff
@@ -35,7 +35,7 @@ class PayloadViolations extends DataProducerPluginBase implements DataProducerPl
    * @return null|\Drupal\social_graphql\GraphQL\ViolationInterface[]
    *   The violations for this payload or null if there are none.
    */
-  public function resolve(PayloadInterface $payload) {
+  public function resolve(PayloadInterface $payload): ?ViolationInterface {
```
Comment (Kingdutch) [inline modules/custom/social_graphql/src/Plugin/GraphQL/DataProducer/Payload/PayloadViolations.php:38]: This return type isn't correct I think. Looks like the comment return type also wasn't correct.

File: modules/custom/social_graphql/src/TypedData/Definition/PageInfoDefinition.php [hunk 6]
```diff
@@ -13,28 +13,26 @@ class PageInfoDefinition extends ComplexDataDefinitionBase {
...
+      ->setLabel("Has previous page")
+      ->setDescription("Whether the result-set has a previous page");
+
+    $info['startCursor'] = DataDefinition::create('string')
+      ->setLabel("Start cursor")
+      ->setDescription("The cursor of the first result in the result-set");
+
+    $info['endCursor'] = DataDefinition::create('string')
+      ->setLabel("End cursor")
+      ->setDescription("The cursor of the last result in the result-set");
```
Comment (Kingdutch) [inline modules/custom/social_graphql/src/TypedData/Definition/PageInfoDefinition.php:35]: What did removing the `!isset` achieve here? 🤔 Do we know why it was here in the first place? Looks like a small performance optimization?

File: modules/custom/social_graphql/tests/src/Kernel/MediaBridgeFileTest.php [hunk 7]
```diff
@@ -59,8 +60,10 @@ protected function setUp() :void {
     // Give anonymous users permission to access content, so they can view and
     // download public files.
     $anonymous_role = Role::load(Role::ANONYMOUS_ID);
-    $anonymous_role->grantPermission('access content');
-    $anonymous_role->save();
+    if ($anonymous_role instanceof RoleInterface) {
+      $anonymous_role->grantPermission('access content');
+      $anonymous_role->save();
+    }
```
Comment (Kingdutch) [inline modules/custom/social_graphql/tests/src/Kernel/MediaBridgeFileTest.php:66]: Lets make this an assert, because not having the `anonymous` role would mean our Drupal instance is severely broken (or Drupal 11 has change in fun and interesting ways).

File: modules/custom/social_graphql/tests/src/Kernel/SocialGraphQLTestBase.php [hunk 8]
```diff
@@ -42,7 +42,7 @@ protected function setUp() : void {
   /**
    * Asserts that a specific endpoint properly implements pagination.
    *
-   * Uses the standardised pagination test cases to validate that pagination
+   * Uses the stand ardised pagination test cases to validate that pagination
```
Comment (Kingdutch) [inline modules/custom/social_graphql/tests/src/Kernel/SocialGraphQLTestBase.php:45]: Undo? :D

File: phpstan-baseline.neon [hunk 9]
```diff
@@ -3386,54 +3386,14 @@ parameters:
...
-			path: modules/custom/social_graphql/src/GraphQL/EntityConnection.php
-
-		-
-			message: "#^Parameter \\#1 \\$result of method Drupal\\\\social_graphql\\\\GraphQL\\\\ConnectionQueryHelperInterface\\:\\:getLoaderPromise\\(\\) expects array, array\\|int given\\.$#"
-			count: 1
-			path: modules/custom/social_graphql/src/GraphQL/EntityConnection.php
-
-		-
-			message: "#^Method Drupal\\\\social_graphql\\\\Plugin\\\\GraphQL\\\\DataProducer\\\\Payload\\\\PayloadViolations\\:\\:resolve\\(\\) should return array\\<Drupal\\\\social_graphql\\\\GraphQL\\\\ViolationInterface\\>\\|null but returns array\\|null\\.$#"
+			message: "#^Method Drupal\\\\social_graphql\\\\Plugin\\\\GraphQL\\\\DataProducer\\\\Payload\\\\PayloadViolations\\:\\:resolve\\(\\) should return Drupal\\\\social_graphql\\\\GraphQL\\\\ViolationInterface\\|null but returns array\\|null\\.$#"
```
Comment (Kingdutch) [inline phpstan-baseline.neon:3389]: This ignored error is actually a bug now that you made it explicit :D (See also my comment on that function ;-) )

**Review by Kingdutch — CHANGES_REQUESTED** [review]
